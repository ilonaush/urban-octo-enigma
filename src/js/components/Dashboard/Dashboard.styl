@import "~js/vars";


.dashboard
  width 100%
  background-color $gray
  position relative
  overflow scroll
