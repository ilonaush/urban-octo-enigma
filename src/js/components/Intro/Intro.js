import "./Intro.styl";
import React from 'react';

const Intro = ({children, ...props}) => {
    return (
    <div className='intro'>
        {children}
    </div>
    )
};

export default Intro;
