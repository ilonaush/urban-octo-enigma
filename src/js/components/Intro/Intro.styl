@import "~js/vars";

.intro
  background linear-gradient(90deg,$dark-blue 0,$blue 60%,$light-blue)
  min-height 40px
  padding 30px
  text-transform uppercase
  color #fff

