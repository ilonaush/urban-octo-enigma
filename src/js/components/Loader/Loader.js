import React from 'react';
import "./Loader.styl"
import {LOADING_TYPES} from "../../constants";

const Loader = ({loadingType}) => {
    switch(loadingType) {
        case LOADING_TYPES.FEED_CAT:
            return (
                <div className='loader'>
                    <img src="/images/feed-cat-loader.gif" alt=""/>
                </div>
            );
        case LOADING_TYPES.HUG_CAT:
            return (
                <div className='loader'>
                    <img src="/images/hug-cat-loader.gif" alt=""/>
                </div>
            );
        case LOADING_TYPES.WASH_CAT:
            return (
                <div className='loader'>
                    <img src="/images/wash-cat-loader.gif" alt=""/>
                </div>
            );
        default:
            return (
                <div className='loader'>
                    <img src="/images/blue-loader.gif" alt=""/>
                </div>
            );
    }
};

export default Loader;
