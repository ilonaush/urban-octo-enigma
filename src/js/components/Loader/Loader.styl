@import "~js/vars";

.loader
  background-color rgba(255,255,255, .5)
  position absolute
  top 0
  left 0
  display flex
  align-items center
  font-size 50px
  text-align center
  width 100%
  justify-content center
  height 100%
  color $orange
