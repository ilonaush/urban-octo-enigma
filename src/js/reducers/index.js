import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer   from './reducers';
import  { saveStore, loadStore } from '../services/LocalStorageService';

const predefinedStore = loadStore() || {};

const store = createStore(
    reducer,
    predefinedStore

);

store.subscribe(() => {
    saveStore(store)
});

export {
    store, predefinedStore
};
