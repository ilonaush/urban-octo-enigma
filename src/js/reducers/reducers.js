import {ACTIONS_TYPES} from "../constants";

const initialState = {
  loading: false,
};

let reducer = (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS_TYPES.GET_CATS:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.ADD_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.FEED_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.HUG_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.WASH_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.ISSUE_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.SET_LOADING: {
      return {
        ...state,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      }
    }
    default:
      return state;
  }
};

export default reducer;
