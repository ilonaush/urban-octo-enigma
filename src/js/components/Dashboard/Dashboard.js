import React, {Component} from 'react';
import './Dashboard.styl';
import Intro from "../Intro/Intro";
import DashboardContent from "../DashboardContent/DashboardContent";
import {connect} from "react-redux";
import Loader from "../Loader/Loader";

export class Dashboard extends Component {

    constructor(props) {
        super(props);
        this.state = {
            loading: false,
            page: ''
        };
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.loading !== this.props.loading) {
            this.setState({
                loading: nextProps.loading
            })
        }
    }

    componentDidMount() {
        this.setState({
            page: this.getPage(this.props.location.pathname, this.props.pages)
        })
    }

    getPage = (path, pages = []) => {
        return {
            ...pages.find((page) => page.path === path)
        };
    };

    render() {
        const {page, loading} = this.state;
        const {loadingType} = this.props;
        return (
            <div className='dashboard'>
                {loading && <Loader loadingType={loadingType}/>}
                <Intro>{page.title}</Intro>
                <DashboardContent/>
            </div>
        );
    }
}

Dashboard.propTypes = {};


export default connect((state) => ({pages: state.pages, loading: state.loading, loadingType: state.loadingType}))(Dashboard);

