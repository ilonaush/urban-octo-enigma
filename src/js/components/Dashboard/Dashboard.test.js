import React from 'react';
import Dashboard  from './Dashboard';
import {MemoryRouter} from "react-router";
import {Provider} from "react-redux";
import {store} from "../../reducers/index";
import renderer from "react-test-renderer";
import Intro from "../Intro/Intro";
import { shallow, mount, render } from 'enzyme';
import Loader from "../Loader/Loader";



describe('Dashboard', () => {
    it('renders correctly', () => {
        const location = { pathname: '/' };
        const DashboardComponent = renderer.create(<MemoryRouter><Provider store={store}><Dashboard location={location}/></Provider></MemoryRouter>);
        expect((DashboardComponent).toJSON()).toMatchSnapshot();
    });

    it('renders correctly page info', () => {
        const props = {
            location : { pathname: '/add-worker' }
        };
        const DashboardComponent = mount(<MemoryRouter><Provider store={store}><Dashboard location={props.location}/></Provider></MemoryRouter>);
        const pageTitle = DashboardComponent.find(Intro).at(0);
        expect(pageTitle.text()).toEqual('Add worker');
    });

    it('has loader when such prop is submitted', () => {
        const props = {
            loading : true,
            location : { pathname: '/add-worker' }
        };
        const DashboardComponent = mount(<MemoryRouter><Provider store={store}><Dashboard {...props}/></Provider></MemoryRouter>);
        expect(DashboardComponent.containsMatchingElement(Loader)).toBeTruthy();
    })
});

