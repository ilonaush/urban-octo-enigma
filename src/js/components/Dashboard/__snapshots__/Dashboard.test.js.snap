// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders correctly 1`] = `
<div
  className="dashboard"
>
  <div
    className="intro"
  >
    Cats of our farm
  </div>
  <div
    className="d-content"
  />
</div>
`;
